--- a/usr/bin/dh_installdocs
+++ b/usr/bin/dh_installdocs
@@ -213,6 +213,11 @@
 			if (!-d "$tmp/usr/share/doc" && !-l "$tmp/usr/share/doc") {
 				install_dir("$tmp/usr/share/doc");
 			}
+
+			# Automatically add a preinst maintainer's script that will take care of the
+			# left-over doc directory if it exists (it has to be removed explicitly).
+			autoscript($package, "preinst", "preinst-remove-usd-dir", "s/#PACKAGE#/$package/");
+
 			# Create symlink to another documentation directory if
 			# necessary.
 			if (!-d "$tmp/usr/share/doc/$package" &&
